src/test/components/AppLocalizationProvider.test.js

```diff
@@ -14,19 +14,37 @@ import {
 } from 'firefox-profiler/selectors/l10n';
 import { lazilyParsedBundles } from 'firefox-profiler/app-logic/l10n';
 import { requestL10n, receiveL10n } from 'firefox-profiler/actions/l10n';
-import { ReactLocalization } from '@fluent/react';
+import { ReactLocalization, Localized } from '@fluent/react';
+import { coerceMatchingShape } from '../../utils/flow';
+import { waitUntilState } from '../fixtures/utils';
 
 describe('AppLocalizationProvider', () => {
   beforeEach(function() {
-    const response = (({
-      ok: true,
-      status: 200,
-      headers: {
-        get: () => 'text/plain',
-      },
-      text: () => Promise.resolve('resolved'),
-    }: any): Response);
-    window.fetch = jest.fn().mockResolvedValue(response);
+    jest.spyOn(window.navigator, 'languages', 'get').mockReturnValue(['en-US']);
+
+    const fetch = jest.fn().mockImplementation((fetchUrl: string) => {
+      if (fetchUrl === '/locales/en-US/app.ftl') {
+        const response = (({
+          ok: true,
+          status: 200,
+          headers: {
+            get: () => 'text/plain',
+          },
+          text: () => Promise.resolve('test-id = This is an English Text'),
```

Comment:
It would be good to assign this `"This is an English Text"` string to a variable  (possible like this: `` `This is a ${language} text` ``)  and return it as a result of setup function. That way we won't have to hardcode the `expect` in the test block. We will be able to use it like this:
```js
    expect(getByText(translatedText)).toBeTruthy();
```

You can return this variable like we do in other setup functions:
```
return { translatedText };
```